src/Amazon.Lambda.Tools/Commands/UpdateFunctionConfigCommand.cs

```diff
@@ -547,7 +600,152 @@ private bool AreDifferent(IEnumerable<string> source, IEnumerable<string> target
+
+            var request = new UpdateFunctionUrlConfigRequest
+            {
+                FunctionName = functionName,
+                AuthType = newAuthType
+            };
+
+            try
+            {
+                this.FunctionUrlLink = (await this.LambdaClient.UpdateFunctionUrlConfigAsync(request)).FunctionUrl;
+
+                if(oldAuthType != newAuthType)
+                {
+                    if(newAuthType == Amazon.Lambda.FunctionUrlAuthType.NONE)
+                    {
+                        await AddFunctionUrlPublicPermissionStatement(functionName);
+                    }
+                    else
+                    {
+                        await RemoveFunctionUrlPublicPermissionStatement(functionName);
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                throw new LambdaToolsException($"Error updating function url config for function {request.FunctionName}: {e.Message}", LambdaToolsException.LambdaErrorCode.LambdaFunctionUrlUpdate, e);
+            }
+        }
+
+        protected async Task DeleteFunctionUrlConfig(string functionName, FunctionUrlAuthType oldAuthType)
+        {
+            var request = new DeleteFunctionUrlConfigRequest
+            {
+                FunctionName = functionName
+            };
+
+            try
+            {
+                await this.LambdaClient.DeleteFunctionUrlConfigAsync(request);
+                if (oldAuthType == Lambda.FunctionUrlAuthType.NONE)
+                {
+                    await RemoveFunctionUrlPublicPermissionStatement(functionName);
+                }
+
+                this.FunctionUrlLink = null;
+            }
+            catch (Exception e)
+            {
+                throw new LambdaToolsException($"Error deleting function url config for function {request.FunctionName}: {e.Message}", LambdaToolsException.LambdaErrorCode.LambdaFunctionUrlDelete, e);
+            }
+        }
+
+        private async Task AddFunctionUrlPublicPermissionStatement(string functionName)
+        {
+            var request = new AddPermissionRequest
+            {
+                StatementId = LambdaConstants.FUNCTION_URL_PUBLIC_PERMISSION_STATEMENT_ID,
+                FunctionName = functionName,
+                Principal = "*",
+                Action = "lambda:InvokeFunctionUrl",
+                FunctionUrlAuthType = Amazon.Lambda.FunctionUrlAuthType.NONE
+            };
+
+            try
+            {
+                this.Logger.WriteLine("Adding Lambda permission statement to public access for Function Url");
+                await LambdaClient.AddPermissionAsync(request);
+            }
+            catch(Amazon.Lambda.Model.ResourceConflictException)
+            {
+                this.Logger.WriteLine($"Lambda permission with statement id {LambdaConstants.FUNCTION_URL_PUBLIC_PERMISSION_STATEMENT_ID} for public access already exists");
+            }
```

Comment:
I see at many places exceptions are completely hidden from customer, is there way like passing to enable more logging for troubleshooting such as `--diagnostics` flag?

Reply:
The error messages are not hidden from the user. The Logger is the tools way of writing to either the console windows or the output pane in Visual Studio.

Reply:
I mean exception here, this is totally hidden from the customer rather they get a customized message. This is from my past experience with deploy tool when not getting extra diagnostic info, which is helpful to diagnose the underlying issues.

Reply:
This is an unfortunate control flow by exception. I can tell if the permission is already added if I get this exception. Otherwise I have to make another service call to get the resource policy and then parse the JSON to find the statement id then make the call. This keeps the operation to one service call.